### PR TITLE
feat(sdk): Allow parametrization of ui_host in kfp.Client

### PR DESCRIPTION
**Pull request by casassg**
**Description of your changes:**

Sometimes host and UI host may not be the same (use case would be having a custom auth proxy for the ml pipelines API server). Would be nice to inject into the client the ui_host so that URLs generated are correct. 

The idea is to consume this public API and overwrite ui_host it via inheritance. We could do this via overloading `self._uihost` but given thats a private api it would be preferable to surface it as a public API instead.

### CODE REVIEW

File: sdk/python/kfp/_client.py [hunk 1]
```diff
@@ -130,11 +130,11 @@ class Client(object):
   LOCAL_KFP_CONTEXT = os.path.expanduser('~/.config/kfp/context.json')
 
   # TODO: Wrap the configurations for different authentication methods.
-  def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client_id=None, other_client_secret=None, existing_token=None, cookies=None, proxy=None, ssl_ca_cert=None, kube_context=None, credentials=None):
+  def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client_id=None, other_client_secret=None, existing_token=None, cookies=None, proxy=None, ssl_ca_cert=None, kube_context=None, credentials=None, ui_host=None):
```
Comment (chensun) [inline sdk/python/kfp/_client.py:133]: If we were to add a new argument, this needs to be documented above in the class docstring.

**Review by chensun — COMMENTED** [review]
> We could do this via overloading self._uihost but given thats a private api it would be preferable to surface it as a public API instead.

@casassg What do you think about overriding via environment variable?
https://github.com/kubeflow/pipelines/blob/a70f0473b6d380d1dfc0c80c350a8330f5498e22/sdk/python/kfp/_client.py#L137

**Comment by casassg** [comment]
> @casassg What do you think about overriding via environment variable?

We have several KF clusters, which means every time we create the client, we would need to set the environ and then clean it up. Otherwise, new instantiations of the client will use the previous UI host if we dont clean up the environment variable well. (not saying it's not possible, just not the cleanest way and there can be side effects)

I rather do it at the python layer to keep client configuration instance based instead of using global state.

**Comment by casassg** [comment]
Sigh. @chensun can you re-approve. Rebased the PR as it seemed stuck with `tide` check.

**Pull request merged**